src/backend/distributed/shared_library_init.c

```diff
@@ -606,13 +607,21 @@ RegisterCitusConfigVariables(void)
 
 	DefineCustomBoolVariable(
 		"citus.subquery_pushdown",
-		gettext_noop("Enables supported subquery pushdown to workers."),
-		NULL,
+		gettext_noop("Usage of this GUC is highly discouraged, please read the long "
+					 "description"),
+		gettext_noop("When enabled, the planner skips many correctness checks "
+					 "for subqueries and pushes down the queries to shards as-is. "
+					 "It means that the queries are likely to return wrong results "
+					 "unless the user is absolutely sure that pushing down the "
+					 "subquery is safe. This GUC is maintained only for backward "
+					 "compatibility, no new users are supposed to use it. The planner"
+					 "is capable of pushing down as much computation as possible to the "
+					 "shards depending on the query."),
 		&SubqueryPushdown,
 		false,
 		PGC_USERSET,
-		GUC_STANDARD,
-		NULL, NULL, NULL);
```

Comment:
Log message only shows up when the setting is changed via `SET` or `ALTER SYSTEM SET`, not on subsequent startup of pg. I guess this is ok, but when set via a script the message might never been seen. Then again, automating things without doing it by hand first might not be the best anyway.

I don't think it is a biggy.

Funnily enough it also shows after a `pg_reload_conf` when set with `ALTER SYSTEM SET` but during the next command.

```
[local] nilsdijk@postgres:9700-68206=# ALTER SYSTEM SET citus.subquery_pushdown TO on;
NOTICE:  Setting citus.subquery_pushdown flag is discouraged becuase it forces the planner to pushdown certain queries, skipping relevant correctness checks.
DETAIL:  When enabled, the planner skips many correctness checks for subqueries and pushes down the queries to shards as-is. It means that the queries are likely to return wrong results unless the user is absolutely sure that pushing down the subquery is safe. This GUC is maintained only for backward compatibility, no new users are supposed to use it. The planner is capable of pushing down as much computation as possible to the shards depending on the query.
ALTER SYSTEM
Time: 12.552 ms
[local] nilsdijk@postgres:9700-68206=# SELECT pg_reload_conf();
┌────────────────┐
│ pg_reload_conf │
├────────────────┤
│ t              │
└────────────────┘
(1 row)

Time: 22.569 ms
[local] nilsdijk@postgres:9700-68206=# SELECT 1;
NOTICE:  Setting citus.subquery_pushdown flag is discouraged becuase it forces the planner to pushdown certain queries, skipping relevant correctness checks.
DETAIL:  When enabled, the planner skips many correctness checks for subqueries and pushes down the queries to shards as-is. It means that the queries are likely to return wrong results unless the user is absolutely sure that pushing down the subquery is safe. This GUC is maintained only for backward compatibility, no new users are supposed to use it. The planner is capable of pushing down as much computation as possible to the shards depending on the query.
┌──────────┐
│ ?column? │
├──────────┤
│        1 │
└──────────┘
(1 row)

Time: 1.412 ms
```

The relation with `pg_reload_conf` was not directly clear to me.

Since `pg_reload_conf` can be called -and I know of installations where it is repeatedly called- from any backend the warning shows up during unsuspected times. This might be good to put attention to it though, however I think it is more noise than useful after a `pg_reload_conf`.

Any chance we could suppress the message when set as the effect of a `pg_reload_conf`?

Reply:
ops, thanks for pointing out this. I pushed one more commit to only notice when the value changes: https://github.com/citusdata/citus/pull/4124/commits/8ecd54bfea8a0c4ae3076f0a75254345b4954dce